### PR TITLE
Feature/consume data from apa

### PR DESCRIPTION
**Pull request by simaob**
Adds a service with two methods to consume data from APA's API.

* `water_quality`: fetches the updated water quality values for different beaches;
* `occupation`: fetches occupation data for portuguese beaches;

You'll need to run `rails db:migrate` as I added a new field to the BeachConfiguration model to hold `water_quality_updated_at` info.

Run the services with the following rails tasks:

`bundle exec rails apa_fetcher:water_quality`
`bundle exec rails apa_fetcher:occupation`

We'll need to set a schedule for these on the server.

### CODE REVIEW

**Comment by simaob** [comment]
Added jobs, and scheduling.

**Comment by simaob** [comment]
rebased with develop.

File: Gemfile [hunk 1]
```diff
@@ -91,6 +91,9 @@ gem 'kaminari'
 # used to decode html in some imported strings
 gem 'htmlentities'
 
+# consume APIs
+gem 'httparty'
```
Comment (santostiago) [inline Gemfile:95]: @simaob do we need this? can't we just use the `Net` module?

File: app/jobs/occupation_update_job.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class OccupationUpdateJob < ApplicationJob
+  queue_as :occupation
+
+  def perform
+    Sidekiq.logger.debug('Starting occupation update')
+    ApaFetcher.occupation
+    closing_time = Time.zone.parse('21:00')
+    next_update = if Time.current.now >= closing_time
+                    Time.zone.parse('8:00') + 1.day
```
Comment (santostiago) [inline app/jobs/occupation_update_job.rb:9]: @simaob these values could be constants (and maybe start earlier than 8am... 6am would be better, I guess)

File: app/jobs/water_quality_update_job.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class WaterQualityUpdateJob < ApplicationJob
+  queue_as :water_quality
+
+  def perform
+    Sidekiq.logger.debug('Starting water quality update')
+    ApaFetcher.water_quality
+    next_update = Time.zone.parse('9:00') + 1.day
```
Comment (santostiago) [inline app/jobs/water_quality_update_job.rb:7]: @simaob same as the previous comment.

File: app/services/importer/import_beaches.rb [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    def cleanup
+      Store.where(store_type: :beach).delete_all
+    end
+
+    def import_beaches_apa
+      data = self.class.get('/perfil.json')&.body
+      JSON.parse(data)['features'].each do |item|
+        beach = item['attributes']
+        store = Store.find_or_initialize_by(name: beach['praia'], store_type: :beach)
```
Comment (santostiago) [inline app/services/importer/import_beaches.rb:22]: @simaob Shouldn't we just update the beaches?

File: app/services/importer/import_beaches.rb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            wc: csv[16],
+            showers: csv[17],
+            accessibility: csv[18],
+            garbage_collection: csv[19],
+            cleaning: csv[20],
+            info_panel: csv[21],
+            restaurant: csv[22],
+            parking: csv[23],
+            parking_spots: csv[24],
+            season_start: season_dates ? Date.parse(season_dates.first) : nil,
```
Comment (santostiago) [inline app/services/importer/import_beaches.rb:88]: @simaob let's put some default values here: 6/6 to 30/08

File: Gemfile (hunk 1 above)
Comment (simaob) [reply Gemfile:95]: I love this gem, let me use it. I love to party!

File: app/services/importer/import_beaches.rb (hunk 4 above)
Comment (simaob) [reply app/services/importer/import_beaches.rb:22]: that's why we are filtering by `store_type: :beach`

**Pull request merged**